test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -40,6 +40,31 @@ namespace Npgsql.Tests
 {
     public class ConnectionTests : TestBase
     {
+
+        [OneTimeSetUp]
+        public void CreateTestFile()
+        {
+            // set up pgpass file with connection credentials
+            const string content = "*:*:*:npgsql_tests:npgsql_tests";
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            var directory = Path.GetDirectoryName(pgpassFile);
+            if (directory != null)
+            {
+                Directory.CreateDirectory(directory);
+            }
+            File.WriteAllText(pgpassFile, content);
+        }
+
+        [OneTimeTearDown]
+        public void DeleteTestFile()
+        {
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            if (File.Exists(pgpassFile))
+            {
+                File.Delete(pgpassFile);
```

Comment:
I've implemented your feedback for the main functionality, but I'm a little confused about this feedback on the tests. Agree completely about what you're saying with destroying the proper pgpass file... I can't believe I didn't realize it before. Are you suggesting that I should go ahead and delete the tests or make it less invasive? If you still see value in the tests, then I will fix them up. While I was adding support for mac/linux file paths, I added code to respect the PGPASSFILE environment variable. That might be a more suitable way to accomplish this. I can store the current value (if set) for the environment variable, create a new temp file with the mock data (adjusting it to use proper username, password, etc from the base class), and then run temporarily with the environment variable pointing to the temp file. In the teardown, I can just restore or delete the environment variable to put it back the way it was. Please let me know your thoughts about that, and I can fix (or delete) the test classes and get the revised code to you.